### ISSUE TITLE
CGAL plugin won't build with image plugin activated

### ISSUE THREAD

**Issue by bcarrez** [sofa-framework/sofa#352]
There are some unresolved externals in MeshGenerationFromImage.inl in the CGAL plugin. 
```Undefined symbols for architecture x86_64:
  "_initImage()", referenced from:
      cgal::MeshGenerationFromImage<sofa::defaulttype::StdVectorTypes<sofa::defaulttype::Vec<3, double>, sofa::defaulttype::Vec<3, double>, double>, sofa::defaulttype::Image<unsigned char> >::update() in MeshGenerationFromImage.cpp.o
      cgal::MeshGenerationFromImage<sofa::defaulttype::StdVectorTypes<sofa::defaulttype::Vec<3, float>, sofa::defaulttype::Vec<3, float>, float>, sofa::defaulttype::Image<unsigned char> >::update() in MeshGenerationFromImage.cpp.o
  "_readImage(char const*)", referenced from:
      cgal::MeshGenerationFromImage<sofa::defaulttype::StdVectorTypes<sofa::defaulttype::Vec<3, double>, sofa::defaulttype::Vec<3, double>, double>, sofa::defaulttype::Image<unsigned char> >::update() in MeshGenerationFromImage.cpp.o
      cgal::MeshGenerationFromImage<sofa::defaulttype::StdVectorTypes<sofa::defaulttype::Vec<3, float>, sofa::defaulttype::Vec<3, float>, float>, sofa::defaulttype::Image<unsigned char> >::update() in MeshGenerationFromImage.cpp.o
  "ImageIO_alloc(unsigned long)", referenced from:
      cgal::MeshGenerationFromImage<sofa::defaulttype::StdVectorTypes<sofa::defaulttype::Vec<3, double>, sofa::defaulttype::Vec<3, double>, double>, sofa::defaulttype::Image<unsigned char> >::update() in MeshGenerationFromImage.cpp.o
      cgal::MeshGenerationFromImage<sofa::defaulttype::StdVectorTypes<sofa::defaulttype::Vec<3, float>, sofa::defaulttype::Vec<3, float>, float>, sofa::defaulttype::Image<unsigned char> >::update() in MeshGenerationFromImage.cpp.o
  "CGAL::Image_3::private_read(point_image*)", referenced from:
      cgal::MeshGenerationFromImage<sofa::defaulttype::StdVectorTypes<sofa::defaulttype::Vec<3, double>, sofa::defaulttype::Vec<3, double>, double>, sofa::defaulttype::Image<unsigned char> >::update() in MeshGenerationFromImage.cpp.o
      cgal::MeshGenerationFromImage<sofa::defaulttype::StdVectorTypes<sofa::defaulttype::Vec<3, float>, sofa::defaulttype::Vec<3, float>, float>, sofa::defaulttype::Image<unsigned char> >::update() in MeshGenerationFromImage.cpp.o
ld: symbol(s) not found for architecture x86_64
```

This file is only included if the image plugin is activated, thus making this error appear only in specific cases.

Moreover, since CGAL seems not tested on CI, this error might be quite old, as it has passed under the radar so far.

It has been observed on OS X, and Linux.

______________________________________________________

Suggested labels:

**Comment by bcarrez** [comment]
unresolved externals fixed by PR #351 
CI still needs CGAL...

**Comment by guparan** [comment]
CGAL is now enabled on CI for configs mac_clang-3.4_options, ubuntu_gcc-5.4_options, ubuntu_clang-3.8_options and ubuntu_clang-3.8_options-debug.
Thus, I close this issue :-)
